src/bidiMapper/modules/context/BrowsingContextImpl.ts

```diff
@@ -419,8 +391,9 @@ export class BrowsingContextImpl {
       if (this.id !== params.frame.id) {
         return;
       }
-      this.#url = params.frame.url + (params.frame.urlFragment ?? '');
-      this.#pendingNavigationUrl = undefined;
+      const url = params.frame.url + (params.frame.urlFragment ?? '');
+      this.#url = url;
```

Comment:
In a follow up would it make sense to refactor the Url to come from the navigation tracking as it knows when its committed and when not.
So we change 
```ts
get #url(){
  return this.#navigationTracker.url;
}
```

Reply:
I guess it should come from document tracker instead when it is implemented.